lib/fennec/evaluator/allocate/request.ex

```diff
@@ -85,12 +85,14 @@ defmodule Fennec.Evaluator.Allocate.Request do
   end
 
   defp verify_unknown_attributes(params, state) do
-    case Params.get_attrs(params) do
-      [] ->
-        {:continue, params, state}
+    with u  <- Params.get_attr(params, Attribute.Username),
+         r  <- Params.get_attr(params, Attribute.Realm),
+         [] <- Params.get_attrs(params) -- [u, r] do
+      {:continue, params, state}
+    else
       _ ->
         {:error, %Attribute.ErrorCode{code: 420}}
```

Comment:
Hm, I see. From my understanding of RFC, unexpected attributes should be just discarded. However if there are attributes which we don't **know** (i.e. cannot decode in our case), they should be returned with this error code.